editor/src/components/canvas/event-helpers.test-utils.tsx

```diff
@@ -694,3 +695,13 @@ export function switchDragAndDropElementTargets(
     fireEvent(targetElement, makeDragEvent('dragover', targetElement, endPoint, fileList))
   })
 }
+
+export async function expectSingleUndoStep<T>(
+  editor: EditorRenderResult,
+  action: () => Promise<T>,
+): Promise<T> {
+  const undoCount = editor.getUndoCount()
+  const result = await action()
+  expect(editor.getUndoCount()).toEqual(undoCount + 1)
+  return result
+}
```

Comment:
The function itself is fine, but this file is the wrong place for it. I _think_ `ui-jsx.test-utils.tsx` would be better 🤔 (unless you can find a better place)

Reply:
I moved it to `utils.test-utils.ts` in [e461839](https://github.com/concrete-utopia/utopia/pull/3178/commits/e461839dff9fa75dd25902d49ef5e623b60e7f1d)